Formula/r.rb

```diff
@@ -125,6 +121,10 @@ def install
       s.gsub!(/^CPPFLAGS =.*/, "\\0 -I#{HOMEBREW_PREFIX}/include")
       s.gsub!(/^LDFLAGS =.*/, "\\0 -L#{HOMEBREW_PREFIX}/lib")
       s.gsub!(/.LDFLAGS =.*/, "\\0 $(LDFLAGS)")
+      # Avoid implicit declarations producing compile-time errors, as seen in
+      # Xcode 12+. This used to be necessary to build R itself, but now should
```

Comment:
Is this reported upstream somewhere?

Reply:
It [was reported for the 4.0.x series](https://bugs.r-project.org/bugzilla/show_bug.cgi?id=18024) by @fxcoudert at the time the original workaround was added to the formula. I understand mainline R and that branch were subsequently patched, although the bug remains open.

Reply:
The suggestion to retain the flag to assist with building packages is [also documented in the R manuals](https://cran.r-project.org/doc/manuals/r-release/R-admin.html#macOS-packages).

Reply:
That reads like people should notify package developers instead of Homebrew making this change for all of them.

Reply:
CRAN is normally very proactive in pushing package developers to adapt to changes in common build environments, but my reading of this actually doesn't suggest that – they seem to be saying that it's not reasonable for package devs to anticipate this. But none of my packages are affected, so I don't know what communication there has been about it.

On the other hand, I've just checked the "official" binary release of R for macOS from CRAN, and it does not set this flag in its `Makeconf`.

My initial approach to this change was to simply delete the `ENV.append` line (and comment) and leave it at that, but I then reinstated it in `Makeconf` to be a little more cautious, since there is R documentation to back this up, and this flag has been part of the formula for over two years. I'm happy to follow whichever route you, as the Homebrew maintainers, prefer – either seems reasonable to me.

Reply:
Let's follow upstream: if they ship without it, let's remove it.

I introduced it in 402d39aeaf88c72a1630f1febb6d0e8ef6fee305 more than two years ago, I think it's fair enough

Reply:
Fair enough! This was my first Homebrew PR so I was tending towards caution 🙂. Now removed entirely.